src/routes/mod.rs

```diff
@@ -107,6 +107,8 @@ pub enum ApiError {
     CustomAuthenticationError(String),
     #[error("Invalid Input: {0}")]
     InvalidInputError(String),
+    #[error("Validation Error: {0}")]
+    ValidationError(String),
```

Comment:
Don't create a new error type, I think `InvalidInputError` works fine for this case.